jubilant/secrettypes.py

```diff
@@ -0,0 +1,143 @@
+"""Dataclasses that contain parsed output from juju secrets."""
+
+from __future__ import annotations
+
+import dataclasses
+import datetime
+import enum
+from typing import Dict, Generic, TypedDict, TypeVar
+
+from typing_extensions import Required
+
+
+class SecretRotateCadence(enum.Enum):
+    """Secret rotation policies."""
+
+    NEVER = 'never'
+    HOURLY = 'hourly'
+    DAILY = 'daily'
+    WEEKLY = 'weekly'
+    MONTHLY = 'monthly'
+    QUARTERLY = 'quarterly'
+    YEARLY = 'yearly'
+
+
+class SecretAccessScope(enum.Enum):
+    """Secret access scopes."""
+
+    UNIT = 'unit'
+    APPLICATION = 'application'
```

Comment:
Given our renames of "app" to "application" throughout Jubilant, I suggest:

```suggestion
    APP = 'application'
```

Reply:
done